kolibri/core/utils/cache.py

```diff
@@ -11,3 +13,85 @@ def __get_process_cache():
 
 
 process_cache = SimpleLazyObject(__get_process_cache)
+_namespace_locks = {}
+
+
+class NamespacedCacheProxy(BaseCache):
```

Comment:
@rtibbles I created this proxy class to generically handle tracking cache keys and clearing them.